### DISCUSSION TITLE
Checking content of saved interest groups

### DISCUSSION THREAD

**Discussion by vincent-grosbois** [privacysandbox/privacy-sandbox-dev-support#11] [General] [answered]
Hello!
Is there a way to list all interest group that were saved to my browser?
It seems that there is no API function in the browser to do this
I know that the Interest Groups are stored by the browser in some kind of database, is there a way to find its location and open it?
Currently this is otherwise very hard to troubleshoot interest groups etc

thanks in advance!

**Comment by samdutton** [comment, marked as answer]
Hi Vincent

> Is there a way to list all interest group that were saved to my browser?

There isn't a JavaScript API for this: interest group memberships are write-only. The only way an interest group can manifest itself is by winning an auction and showing an ad — and even then, the ad appears inside a fenced frame, so that the surrounding page cannot learn interest group membership.

> I know that the Interest Groups are stored by the browser in some kind of database, is there a way to find its location and open it? Currently this is otherwise very hard to troubleshoot interest groups etc

We're looking into ways to make FLEDGE data available locally for debugging — I'll update this thread with more information about how this will be done.

**Reply by samdutton** [reply]
Marking this as answered, but feel free to reopen or reply.

**Reply by vincent-grosbois** [reply]
Hi @samdutton I'm not sure what you mean by "answered", did the status change since June ?
ie are there a way to check the FLEDGE data once a user was added to an interest group?

**Reply by hirenshah9f** [reply]
A javascript API to query interest groups, albeit on limited basis, would help automate testing. Something as simple and limited as is interest group x joined for owner y in last z miliseconds would help. Interest groups can make themselves available for such a query if needed.

**Reply by samdutton** [reply]
Edit: We will look more into it, and get back to you.
